components/calendar/calendar-header.component.ts

```diff
@@ -101,6 +101,9 @@ export class NzCalendarHeaderComponent implements OnInit {
   ngOnInit(): void {
     this.setUpYears();
     this.setUpMonths();
+    this.i18n.localeChange.subscribe(() => {
+      this.setUpMonths();
+    });
```

Comment:
`this.i18n.localeChange` is `BehaviorSubject`, so `setUpMonths` will be triggered twice